File: exercism.io/javascript/isogram/isogram.js
/**
 * Created by mewx on 19/12/16.
 */

var Isogram = function(str) {
    this.isLetter = function(str) {
        return str.length === 1 && str.match(/[a-z]/i);
    };

    this.isIsogram = function() {
        str = str.toLowerCase();
        if (str.length <= 1) return true;
        for (var i = 1; i < str.length; i ++) {
            // if (!this.isLetter(str[i])) continue;
            if (str[i] == '-' || str[i] == ' ') continue;
            if (str.substring(0, i).includes(str[i]))
                return false;
        }
        return true;
    };
};

module.exports = Isogram;

File: exercism.io/go/isogram/isogram.go
package isogram

import (
	"unicode"
)

// IsIsogram checks whether the input string is an isogram.
func IsIsogram(str string) bool {
	mem := make(map[rune]bool)
	for _, originalR := range str {
		r := unicode.ToLower(originalR)
		if !unicode.IsLetter(r) {
			continue
		}

		if mem[r] {
			return false
		}
		mem[r] = true
	}

	return true
}
